### ISSUE TITLE
Migrate discord from clj to cljs

### ISSUE THREAD

**Issue by azureus89** [224-1/xtnt-micro#44]
@kattakavya New build for discord, just like trello
once @ParikshithMohan is done with his documentation and I pull the discord ui clojure code

**Comment by azureus89** [comment]
also once @sahabaishakhi is done with documentation

**Comment by azureus89** [comment]
@kattakavya you can start with this now ... @ParikshithMohan @sahabaishakhi both of your documentation will be put to test

**Comment by azureus89** [comment]
be available to help out, and if gaps in documentation are exposed, make the changes locally, in whatever branch you are working on, no need for separate documentation branch for this

**Comment by kattakavya** [comment]
**Context:** Migration from clj to cljs, able to create multiple builds
**Issue:** Not able to render the two functions at a time.
**Logs:** no errors found in repl and console.
![image](https://user-images.githubusercontent.com/46553145/67227171-53b03d80-f454-11e9-8c1e-dc6cef92f9b8.png)

when I given single function,it is able to render.
![image](https://user-images.githubusercontent.com/46553145/67227337-b0135d00-f454-11e9-9f93-3c87bf0506da.png)

Conclusion: I didn't find any syntax error in repl and in console also,so trying to get what went wrong.

**Comment by sahabaishakhi** [comment]
@kattakavya you seem to have replaced all of the underscores with dashes but didn't do so in the css file. so do that and also do the same in clj file of @ParikshithMohan

**Comment by azureus89** [comment]
@kattakavya resolved?

**Comment by kattakavya** [comment]
@azureus89 No, @ParikshithMohan  have been calling function inside a function,those functions are not getting rendered,if I write separate function i can able to execute it.
**for example**: In code, the function container-2  is defined there he have been calling the child functions
 (container-2-head)
(channel-list)
(container-2-footer)
if I render only container-2 function it is not executing
**Error in console**
![image](https://user-images.githubusercontent.com/46553145/67278396-886cd500-f4e6-11e9-9f03-a3ca0210f43f.png)

**code**
(defn discord-ui []
  [:div
   (container-2)])

 (reagent/render-component
[discord-ui]
 (. js/document (getElementById "app")))

I can able to render if I write separate function
![image](https://user-images.githubusercontent.com/46553145/67278893-8c4d2700-f4e7-11e9-93de-814ee14de769.png)

**code**
(defn discord-ui []
  [:div
   (container-1)
   (container-2-head)
   (container-2-footer)])
(reagent/render-component
[discord-ui]
 (. js/document (getElementById "app")))

**Comment by azureus89** [comment]
@kattakavya that is expected behaviour
you have to replicate all the elements

calling functions inside functions is the advantage of writing code in a functional language
it maybe unusual for front end code to work that way, but that's what we are trying to do differently
it is very vital part of clojurescript and more specifically react / reactive coding

**Comment by azureus89** [comment]
@kattakavya wait I didnt understand what you are saying I think

**Comment by kattakavya** [comment]
@azureus89 It is advantage of writing function inside the function true, coming to clojure script the syntax would be different for writing function inside function i think..so trying to find out the syntax for writing it.

**Comment by azureus89** [comment]
yes .. issue seems to be around the use of the img tag in one of the functions
looks like something is being nested inside an img tag somewhere
whether typo or something parikshith did, and was okay in html, but not something react appreciates

**Comment by azureus89** [comment]
understood what you were saying now ... the second error also should have preseneted in the format discussed ... it helps avoid confusion ... in general also what you could have done better is come up with a hypothesis ... in the first case it was hard and needed more experimentation .. but finally when you produced errors, you could have come up with a hypothesis better ... the error says its about img tag ... and that img cannot have children
ie. something is being nested inside an img tag in one of the functions that discord-ui is recursively calling
which react does not like
so hypothesis / highlight was very easy to find ^

**Comment by azureus89** [comment]
from the error I can also make out that in your split function where you are calling the functions individually instead of calling container-2 ... you are not calling channel-list ... 
if you call channel list does it throw up error, is that why it is left out?

**Comment by azureus89** [comment]
show me dev console when channel-list is included in the calls, if it throws and error, then the error is wrongly reported -- it is not because of anything to do with nesting function calls / calling syntax ... it is because of channel-list

i want you to report in the error format prescribed because it is a common mistake to jump to conclusions and hypothesis ... force yourself to have all the data in front of you, before you narrow down and come up with a hypothesis ... because errors reported with the wrong hypothesis / conclusions waste a lot of time

generally, in any science / activity ... if you start with a conclusion you can always show evidence to prove the conclusion .. which is what has happened here ... avoid that, it is the wrong approach to engineering

**Comment by azureus89** [comment]
> @azureus89 It is advantage of writing function inside the function true, coming to clojure script the syntax would be different for writing function inside function i think..so trying to find out the syntax for writing it.

syntax for calling functions is not different ... you have gone down the wrong road because of what I described above ... dont jump to conclusions without enough testing ... 

what is different is : react does not allow img tags to have children - in parikshiths code there was no react ... its not a clojure/clojurescrtipt/syntax/function calling problem

**Comment by kattakavya** [comment]
Context: migration from clj-cljs discord-ui
Issue: Error causing while i include channel-list
Logs in repl:
[Figwheel:WARNING] Compile Warning   src\cljs\xtnt_micro\discordui.cljs   line:222  column:4
 Wrong number of args (0) passed to xtnt-micro.discordui/channel-list
  217  (defn discord-ui []
  218    [:div
  219     (container-1)
  220     (container-2-head)
  221     (container-2-footer)
  222     (channel-list)
          ^---
  223     ])
  224  
  225  ;;(defn get-app-element []
  226  ;;  (gdom/getElement "app"))
  227  

[Figwheel] Successfully compiled build dscrd to "target\public\cljs-out\dscrd-main.js" in 1.422 seconds.

Highlights: due to wrong args pass to the channel-list function
Hypothesis: channel-list contains the img tag and unordered list,should change the syntax i guess
Conclusion: first i will test by correcting the syntax of channel-list, later if the error repeats,i will try to change the syntax of img tag.

**Comment by azureus89** [comment]
just change img tag to div ... img tag cant have children

**Comment by kattakavya** [comment]
@azureus89 changed to div tag,

**code**
(defn channel-list [arg]
  (let [current (-> @state :server (@active-server) arg)]
    (map #(identity [:div.c2-rooms
                     [:div#channel-logo [:img {:src (:logo current)}
                                         [:ul [:li (val %)]]]]
                     [:br]]) (-> current :list))))
**Error**
Wrong number of args (0) passed to xtnt-micro.discordui/channel-list

  218  (defn discord-ui []
  219    [:div
  220     (container-1)
  221     (container-2-head)
  222     (container-2-footer)
  223     (channel-list)
          ^---
  224     ])
  225  
  226  ;;(defn get-app-element []
  227  ;;  (gdom/getElement "app"))
  facing same error.

**Comment by azureus89** [comment]
@kattakavya channel-list is called with an arg ... please check original code

**Comment by azureus89** [comment]
this is an expected error when you are calling without argument
img tag to div tag is not related to this but previous error 

try to run container-2 now instead of this list
check container-2 to see how channel-list is being called, its quite clear

**Comment by ParikshithMohan** [comment]
@kattakavya U sure u want to use container-2-head and container-2-footer in the main function?...because those two are already used inside container-2...so container-2 alone can be used instead of two extra functions.

channel-list requires an argument to be passed...preferably from the atom....something like this - (channel-list :rooms)

**Comment by kattakavya** [comment]
@azureus89 container-2 is displaying without errors but styling got changed it is not same like in clj,why excatly?
![image](https://user-images.githubusercontent.com/46553145/67368883-b3fac880-f595-11e9-8baa-1d51a6390889.png)
should i go-head and change the css-styling

**Comment by azureus89** [comment]
> @kattakavya U sure u want to use container-2-head and container-2-footer in the main function?...because those two are already used inside container-2...so container-2 alone can be used instead of two extra functions.
> 
> channel-list requires an argument to be passed...preferably from the atom....something like this - (channel-list :rooms)

source of the issue is the nesting of things inside an img tag -- img tags aren't meant for that ... thats a change that needs to be done ..

**Comment by azureus89** [comment]
> @azureus89 container-2 is displaying without errors but styling got changed it is not same like in clj,why excatly?
> ![image](https://user-images.githubusercontent.com/46553145/67368883-b3fac880-f595-11e9-8baa-1d51a6390889.png)
> should i go-head and change the css-styling

please show code also
because this is wrong:
`(defn channel-list [arg]
(let [current (-> @State :server (@active-server) arg)]
(map #(identity [:div.c2-rooms
[:div#channel-logo [:img {:src (:logo current)}
[:ul [:li (val %)]]]]
[:br]]) (-> current :list))))`

**Comment by kattakavya** [comment]
@azureus89 

**code:**
(defn channel-list [arg]
  (let [current (-> @state :server (@active-server) arg)]
    (map #(identity [:div.c2-rooms
                     [:div#channel-logo [:img {:src (:logo current)}]
                                         [:ul [:li (val %)]]
                     [:br]]]) (-> current :list))))

I'm calling container-2 itself as single function.

(defn discord-ui []
  [:div
   (container-1)
   (container-2)
   (main-header)])

(reagent/render-component
[discord-ui]
 (. js/document (getElementById "app")))

**Comment by azureus89** [comment]
> [:div#channel-logo [:img {:src (:logo current)}]
> [:ul [:li (val %)]]

yes, with this styling you will have to change the css
@ParikshithMohan you have to make this change locally on your clojure branch and push the css to make it work
please do that and send PR -- need to merge to master asap
kavya will inherit the styles

**Comment by kattakavya** [comment]
**Context:** migration from clj-cljs discord-ui
**Issue:** In container-3 we have P tag which has styling property,i think because of that and because of string, container-3 is not executing.
**Logs:** No errors found in repl. Errors in console,
The above error occurred in the <p> component:
    in p (created by xtnt_micro.discordui.discord_ui)
    in div (created by xtnt_micro.discordui.discord_ui)
    in div (created by xtnt_micro.discordui.discord_ui)
    in div (created by xtnt_micro.discordui.discord_ui)
    in div (created by xtnt_micro.discordui.discord_ui)
    in div (created by xtnt_micro.discordui.discord_ui)
    in div (created by xtnt_micro.discordui.discord_ui)
    in div (created by xtnt_micro.discordui.discord_ui)
    in div (created by xtnt_micro.discordui.discord_ui)
    in xtnt_micro.discordui.discord_ui

Consider adding an error boundary to your tree to customize error handling behavior.
Visit https://fb.me/react-error-boundaries to learn more about error boundaries.
logCapturedError @ react-dom.inc.js:10089
logError @ react-dom.inc.js:10128
commitErrorLogging @ react-dom.inc.js:10341
commitAllLifeCycles @ react-dom.inc.js:11905
callCallback @ react-dom.inc.js:140
invokeGuardedCallbackDev @ react-dom.inc.js:178
invokeGuardedCallback @ react-dom.inc.js:227
commitRoot @ react-dom.inc.js:12040
completeRoot @ react-dom.inc.js:12948
performWorkOnRoot @ react-dom.inc.js:12898
performWork @ react-dom.inc.js:12816
performSyncWork @ react-dom.inc.js:12793
requestWork @ react-dom.inc.js:12693
scheduleWorkImpl @ react-dom.inc.js:12568
scheduleWork @ react-dom.inc.js:12528
scheduleRootUpdate @ react-dom.inc.js:13156
updateContainerAtExpirationTime @ react-dom.inc.js:13184
updateContainer @ react-dom.inc.js:13211
ReactRoot.render @ react-dom.inc.js:16647
legacyRenderSubtreeIntoContainer @ react-dom.inc.js:17081
render @ react-dom.inc.js:17121
reagent$dom$render_comp @ dom.cljs:20
(anonymous) @ dom.cljs:44
reagent$dom$render @ dom.cljs:31
(anonymous) @ dom.cljs:39
reagent$dom$render @ dom.cljs:31
(anonymous) @ core.cljs:74
reagent$core$render @ core.cljs:65
(anonymous) @ discordui.cljs:234
react-dom.inc.js:59 Uncaught Invariant Violation: The `style` prop expects a mapping from style properties to values, not a string. For example, style={{marginRight: spacing + 'em'}} when using JSX.
    in p (created by xtnt_micro.discordui.discord_ui)
    in div (created by xtnt_micro.discordui.discord_ui)
    in div (created by xtnt_micro.discordui.discord_ui)
    in div (created by xtnt_micro.discordui.discord_ui)
    in div (created by xtnt_micro.discordui.discord_ui)
    in div (created by xtnt_micro.discordui.discord_ui)
    in div (created by xtnt_micro.discordui.discord_ui)
    in div (created by xtnt_micro.discordui.discord_ui)
    in div (created by xtnt_micro.discordui.discord_ui)
    in xtnt_micro.discordui.discord_ui
    at invariant (http://localhost:9500/cljs-out/dscrd/cljsjs/react-dom/development/react-dom.inc.js:59:15)
    at assertValidProps (http://localhost:9500/cljs-out/dscrd/cljsjs/react-dom/development/react-dom.inc.js:14370:63)
    at setInitialProperties$1 (http://localhost:9500/cljs-out/dscrd/cljsjs/react-dom/development/react-dom.inc.js:15551:3)
    at finalizeInitialChildren (http://localhost:9500/cljs-out/dscrd/cljsjs/react-dom/development/react-dom.inc.js:16817:5)
    at completeWork (http://localhost:9500/cljs-out/dscrd/cljsjs/react-dom/development/react-dom.inc.js:9823:19)
    at completeUnitOfWork (http://localhost:9500/cljs-out/dscrd/cljsjs/react-dom/development/react-dom.inc.js:12117:20)
    at performUnitOfWork (http://localhost:9500/cljs-out/dscrd/cljsjs/react-dom/development/react-dom.inc.js:12277:14)
    at workLoop (http://localhost:9500/cljs-out/dscrd/cljsjs/react-dom/development/react-dom.inc.js:12289:26)
    at renderRoot (http://localhost:9500/cljs-out/dscrd/cljsjs/react-dom/development/react-dom.inc.js:12320:9)
    at performWorkOnRoot (http://localhost:9500/cljs-out/dscrd/cljsjs/react-dom/development/react-dom.inc.js:12895:24)
invariant @ react-dom.inc.js:59
assertValidProps @ react-dom.inc.js:14370
setInitialProperties$1 @ react-dom.inc.js:15551
finalizeInitialChildren @ react-dom.inc.js:16817
completeWork @ react-dom.inc.js:9823
completeUnitOfWork @ react-dom.inc.js:12117
performUnitOfWork @ react-dom.inc.js:12277
workLoop @ react-dom.inc.js:12289
renderRoot @ react-dom.inc.js:12320
performWorkOnRoot @ react-dom.inc.js:12895
performWork @ react-dom.inc.js:12816
performSyncWork @ react-dom.inc.js:12793
requestWork @ react-dom.inc.js:12693
scheduleWorkImpl @ react-dom.inc.js:12568
scheduleWork @ react-dom.inc.js:12528
scheduleRootUpdate @ react-dom.inc.js:13156
updateContainerAtExpirationTime @ react-dom.inc.js:13184
updateContainer @ react-dom.inc.js:13211
ReactRoot.render @ react-dom.inc.js:16647
legacyRenderSubtreeIntoContainer @ react-dom.inc.js:17081
render @ react-dom.inc.js:17121
reagent$dom$render_comp @ dom.cljs:20
(anonymous) @ dom.cljs:44
reagent$dom$render @ dom.cljs:31
(anonymous) @ dom.cljs:39
reagent$dom$render @ dom.cljs:31
(anonymous) @ core.cljs:74
reagent$core$render @ core.cljs:65
(anonymous) @ discordui.cljs:234
console.js:203  [Figwheel] loaded (xtnt_micro.discordui)
**Highlights:** : Uncaught Invariant Violation: The `style` prop expects a mapping from style properties to values, not a string. For example, style={{marginRight: spacing + 'em'}} when using JSX.
**code:** current code which we have.
(defn chat-direct []
  [:div.chat-direct
   [:div#direct-userlogo [:img {:src (-> @messages (@active-server) :chat1 :sender-logo)}]]
   [:div#direct-username [:p {:style (str "color:" (rand-nth (-> @state :color)) ";")}] (-> @messages (@active-server)  :chat1 :sender)]
   [:div#direct-timestamp [:p (-> @messages (@active-server) :chat1 :timestamp)]]
   [:div#direct-message [:p (-> @messages (@active-server) :chat1 :content)]]])

(defn chat-condition1 [] (if (= (-> @messages (@active-server) :chat0 :host first) "GitHub")  (chat-bot)))
(defn chat-condition2 [] (if (= (-> @messages (@active-server) :chat1 :host first) "direct")  (chat-direct)))

;;Whole middle section with messages typing
(defn container-3 []
  [:div.container-3
   [:div.c3-contents
    (chat-condition1)
    (chat-condition2)]
   [:div.messaging
    [:div#plus-c3]
    [:div#type-message [:input {:type "text" :placeholder "Message #xtnt-micro"}]]
    [:div#smiley-c3.messaging-right-icon]
    [:div#gif-c3.messaging-right-icon]
    [:div#gift-c3.messaging-right-icon]]])

;;Member list organization function
(defn members-list [arg]
  (let [current (-> @state :members arg)]
    [:div.members-list (map #(identity [:div.members-content
                                        [:div#member-logo {:src (get-logo %)} [:p {:style (str "color:" (rand-nth (-> @state :color))";")} %]]
                                        [:br]]) (-> current))]))


**Hypothesis**: I think by changing the p tag,can eradicate errors.
**Conclusion:** Is the hypothesis right ?I'm trying by changing the p tag.

**Comment by ParikshithMohan** [comment]
**After adding the div tag:**
![image](https://user-images.githubusercontent.com/24820582/67557117-1092ea80-f732-11e9-93b6-97c884bd7150.png)

Changes made in both clj and css...

**Comment by azureus89** [comment]
@ParikshithMohan thanks .. @kattakavya 
1) clj and css working with the img tag bug fixed
2) any progress on inline-styling syntax? from your error looks like syntax for inline styling is wrong -- it says expecting a map not a string

**Comment by sahabaishakhi** [comment]
@kattakavya I guess I need to add documentation for reagent inline styling, it doesn't take in attributes within attributes by providing them in quotes. instead you need to use nested hashmaps. 
for ex - in clj/hiccup file it must've been implemented in this manner - `[:div.a {:style "width:40px"}]`
whereas in cljs/reagent it should look like - `[:div.a {:style {:width "40px"}}]`

**Comment by kattakavya** [comment]
**Context:** migration from clj to cljs
**Issue:**  Facing issue with the color and with the p tag
**Logs**: no errors in the repl,errors found in the console
react-dom.inc.js:526 Warning: validateDOMNesting(...): <div> cannot appear as a descendant of <p>.
    in div (created by xtnt_micro.discordui.discord_ui)
    in p (created by xtnt_micro.discordui.discord_ui)
    in div (created by xtnt_micro.discordui.discord_ui)
    in div (created by xtnt_micro.discordui.discord_ui)
    in div (created by xtnt_micro.discordui.discord_ui)
    in div (created by xtnt_micro.discordui.discord_ui)
    in div (created by xtnt_micro.discordui.discord_ui)
    in div (created by xtnt_micro.discordui.discord_ui)
    in xtnt_micro.discordui.discord_ui
printWarning @ react-dom.inc.js:526
warning @ react-dom.inc.js:550
validateDOMNesting @ react-dom.inc.js:16424
createInstance @ react-dom.inc.js:16800
completeWork @ react-dom.inc.js:9816
completeUnitOfWork @ react-dom.inc.js:12117
performUnitOfWork @ react-dom.inc.js:12277
workLoop @ react-dom.inc.js:12289
renderRoot @ react-dom.inc.js:12320
performWorkOnRoot @ react-dom.inc.js:12895
performWork @ react-dom.inc.js:12816
performSyncWork @ react-dom.inc.js:12793
requestWork @ react-dom.inc.js:12693
scheduleWorkImpl @ react-dom.inc.js:12568
scheduleWork @ react-dom.inc.js:12528
scheduleRootUpdate @ react-dom.inc.js:13156
updateContainerAtExpirationTime @ react-dom.inc.js:13184
updateContainer @ react-dom.inc.js:13211
ReactRoot.render @ react-dom.inc.js:16647
(anonymous) @ react-dom.inc.js:17066
unbatchedUpdates @ react-dom.inc.js:13003
legacyRenderSubtreeIntoContainer @ react-dom.inc.js:17062
render @ react-dom.inc.js:17121
reagent$dom$render_comp @ dom.cljs:20
(anonymous) @ dom.cljs:44
reagent$dom$render @ dom.cljs:31
(anonymous) @ dom.cljs:39
reagent$dom$render @ dom.cljs:31
(anonymous) @ core.cljs:74
reagent$core$render @ core.cljs:65
(anonymous) @ discordui.cljs:233
react-dom.inc.js:526 Warning: Style property values shouldn't contain a semicolon. Try "#9b59b6: " instead.
    in p (created by xtnt_micro.discordui.discord_ui)
    in div (created by xtnt_micro.discordui.discord_ui)
    in div (created by xtnt_micro.discordui.discord_ui)
    in div (created by xtnt_micro.discordui.discord_ui)
    in div (created by xtnt_micro.discordui.discord_ui)
    in div (created by xtnt_micro.discordui.discord_ui)
    in xtnt_micro.discordui.discord_ui
**Highlights**: 
1.Warning: validateDOMNesting(...): <div> cannot appear as a descendant of <p>.
2.Style property values shouldn't contain a semicolon. Try "#9b59b6: " instead.
**Hypothesis:** for the 2nd warning i have replaced ";" with the some-color, later warring didn't came but here we are using rand function for different colors,if i use give single color it of no use,i think it is not a permanent solution.
**code**
(defn members-list [arg]
  (let [current (-> @state :members arg)]
    [:div.members-list (map #(identity [:div.members-content
                                        [:div#member-logo [:img {:src (get-logo %)}] [:p {:style {str "color:" (rand-nth (-> @state :color))";"}} %]]
                                        [:br]]) (-> current))]))

![image](https://user-images.githubusercontent.com/46553145/67589675-a2244b80-f776-11e9-9e87-2b3de303f7b4.png)

**Conclusion:** I have tried to convert most of things and changed in clj and css too but stuck with 4 things,couldn't able to get rand color for chat members,facing error with p tag and for container-1 the height was given as 100% but it is not taking..full screen,if i'm giving in pixels it is taking the full screen but what is problem for not taking the % and taking in pixels ?
![image](https://user-images.githubusercontent.com/46553145/67590137-d51b0f00-f777-11e9-89d9-916ea82eb80b.png)

**Comment by ParikshithMohan** [comment]
@kattakavya Try putting `{str "color:" (rand-nth (-> @State :color))";"}` inside a separate function and then calling that function in :style. Maybe the colors are not showing up because they were inside the state and not css or hiccup.

**Comment by azureus89** [comment]
come on guys @ParikshithMohan @kattakavya , this is silly think about this ... just because its clojursecript you think its something complicated ... 
` {str "color:" (rand-nth (-> @State :color))";"}`

look at this expression ... this is obviously going to give you an error
this is supposed to be a map ... this is a very basic and silly mistake you are making ... dont know what the solution is, but its quite obvious whats wrong with what you are doing here
this is a map ... if you are going to use a map it has to be a key value pair ... so
`{:color "#445ss"}`   is the expected final format
so
`{:color (rand-nth (-> @State :color))}`
tried that?

**Comment by azureus89** [comment]
@kattakavya one error at a time

**Comment by azureus89** [comment]
> Reagent builds on that and lets you use a ClojureScript map to represent inline styles. The same rules apply: the name of the CSS property is the key and the value of the CSS property is the value. Here's what the above div with inline styles will look like:
> 
> [:div {:style {:margin-left 10
>                :width "15%"}}
>  [:p "Some text"]]

**Comment by kattakavya** [comment]
@azureus89 
![image](https://user-images.githubusercontent.com/46553145/67621866-a5304200-f831-11e9-84fe-2f1f706ae8cc.png)
now all attributes added,i think its matter of styling..i will be working on it.

**Comment by azureus89** [comment]
@kattakavya okay good, next focus only on the left side bar ... 
is your feature branch at latest?

**Comment by kattakavya** [comment]
@azureus89 facing a problem to fix styling part for container-4
![image](https://user-images.githubusercontent.com/46553145/67693152-41d81880-f9c7-11e9-8ff4-edf61f5f9abf.png)

No,facing merge conflict,
![image](https://user-images.githubusercontent.com/46553145/67693262-6fbd5d00-f9c7-11e9-96af-ae7f7f2389ad.png)

**Comment by azureus89** [comment]
@kattakavya doesnt look like merge conflict ... just the last commit on master is called "merge pull request ..."
pull from master to your branch (magit: F u) ... if theres a conflict it will give you an error message ... 
if no conflict, go ahead and push to your feature branch ...
do that, then i can see the code, and help you out with the container-4 issue

**Comment by azureus89** [comment]
@sahabaishakhi have merged into master, can you check this, and if all working can move you and kavya to next issue and close this one

**Comment by azureus89** [comment]
> @kattakavya I guess I need to add documentation for reagent inline styling, it doesn't take in attributes within attributes by providing them in quotes. instead you need to use nested hashmaps.
> for ex - in clj/hiccup file it must've been implemented in this manner - `[:div.a {:style "width:40px"}]`
> whereas in cljs/reagent it should look like - `[:div.a {:style {:width "40px"}}]`

yes plaese add ... all these issues are opportunities to improve and compact internal documentation

**Comment by sahabaishakhi** [comment]
@azureus89 it's working for me
